### ISSUE TITLE
Simultaneous file writes using `putContentsAsync` fail on Windows [test with a newer Firefox]

### ISSUE THREAD

**Issue by mrtcode** [zotero/zotero#2210]
```
Error: Win error 183 during operation rename on file test.txt (Cannot create a file when that file already exists. ) (No stack trace) From previous event: Zotero.File</this.putContentsAsync@chrome://zotero/content/xpcom/file.js:410:14 
```

The "rename" part is probably coming from`OS.File.writeAtomic` using temporary file (`tmpPath`) which is "a little safer". This might be a Firefox bug.

Reproduced with:
```
async test() {
  try {
    await Zotero.File.putContentsAsync('test.txt', 'asdf');
  }
  catch (e) {
    Zotero.debug(e);
    Zotero.crash();
  }
}

for (let i = 0; i < 1000; i++) {
  this.test();
}
```

**Comment by mrtcode** [comment]
1000 simultaneous writes are too much for real-life cases, but on a slow system (old spinning disk, network), I suspect, just two simultaneous calls might be enough.

**Comment by mrtcode** [comment]
Although the user was experiencing a bit different error, and yes there is a chance it was just another program holding the file, but we will see.
```
Stack: Win error 32 during operation delete on file C:\Users\...\storage\6GJKNK7N.zotero-pdf-state (Impossible to open file. The file is used by another process.
)
```

**Comment by dstillman** [comment]
Is there any reason the code changed in 1f634f9 should run simultaneously?

**Comment by mrtcode** [comment]
Yes, pdf-reader can trigger another state update while the first one is being saved, even though we are debouncing the saving. And this, in theory, can happen if the first state saving takes too long or was triggered not immediately (slow system, network file system). Although, I agree, all of this is unlikely, unless the system is extremely slow.

**Comment by dstillman** [comment]
A couple recent reports of the .zotero-pdf-state issue:

https://forums.zotero.org/discussion/94315/error-setstate-with-empty-error-message (from https://github.com/zotero/zotero/issues/2328)
https://forums.zotero.org/discussion/94465/zotero-pdf-reader-raise-error

Do we think that's from this issue or just security software or similar on people's machines?
